### PR TITLE
Feat: Support of converting local models added to `convert-hf-to-gguf-update.py`

### PR DESCRIPTION
**Pull request by EvilFreelancer**
## Description

For one of my projects, I needed to convert a trained BERT model, which was only available on my local filesystem. However, when I attempted the conversion, I encountered an error indicating that my tokenizer was not supported. To resolve this, I added my local model and its filesystem path to the models array, but running the `convert-hf-to-gguf-update.py` script did not update `convert-hf-to-gguf.py` as expected.

The root cause was identified in the `download_model` function. The current implementation only supports downloading models hosted on HuggingFace.

But my fix extends this functionality to also support models stored locally.

## Changes Made

* Modified the `download_model` function to check if the `repo` path is a local directory.
* If `repo` is a local path, the function now copies the required files from the local directory to the appropriate destination.
* If `repo` is a URL, the function retains its original behavior of downloading the files from HuggingFace.

## Example Usage

To convert a local model, you can now add it to the models array of `convert-hf-to-gguf-update.py` with the local filesystem path:

```python
models = [
    {"name": "sbert", "tokt": TOKENIZER_TYPE.WPM, "repo": "./sbert_large_nlu_ru"},
    ...
    ...
]
```

Run it:

```
python convert-hf-to-gguf-update.py <hf_token>
```

And new block will be added to `convert-hf-to-gguf.py`, it will looks like this:

```python
if chkhsh == "eb70cf7b9840b5efff6c673f695000d9a0195852c07a45a3e61718fcbaddf601":
    # ref: ./sbert_large_nlu_ru
    res = "sbert"
```

After than you will be able to convert local model to the GGUF format.

### CODE REVIEW

File: convert-hf-to-gguf-update.py [hunk 1]
```diff
@@ -106,12 +107,26 @@ def download_model(model):
...
         files.append("tokenizer.model")
 
-    for file in files:
-        save_path = f"models/tokenizers/{name}/{file}"
-        if os.path.isfile(save_path):
-            logger.info(f"{name}: File {save_path} already exists - skipping")
-            continue
-        download_file_with_auth(f"{repo}/resolve/main/{file}", token, save_path)
+    if os.path.isdir(repo):
+        # Если repo это путь на файловой системе, копируем директорию
```
Comment (akx) [inline convert-hf-to-gguf-update.py:111]: This comment should preferably be in English, I think :)

Comment (EvilFreelancer) [reply convert-hf-to-gguf-update.py:111]: Opps, sorry :)

Comment (EvilFreelancer) [reply convert-hf-to-gguf-update.py]: Fixed

**Comment by EvilFreelancer** [comment]
Any news about my PR? People need this simple feature :)

**Review by ggerganov — APPROVED** [review]
We can merge after rebasing on latest `master`

**Comment by EvilFreelancer** [comment]
@ggerganov hi! You mean I need to merge `master` branch into this feature branch?

**Comment by ggerganov** [comment]
Yes. The `convert-hf-to-gguf-update.py` file was renamed at some point to `convert_hf_to_gguf_update.py` and so there is a merge conflict preventing the merging of the PR. After you resolve it, we will be able to merge.

**Comment by EvilFreelancer** [comment]
Done, i've also tested this new feature by hands, it works fine.

**Pull request merged**